app/configuration/bundle_parameters.go

```diff
@@ -65,6 +69,67 @@ const (
 
 var systemParameters = map[string]func() (string, error){
 	"sys.host": os.Hostname,
+	"sys.pkg_arch": func() (string, error) {
+		if software.DefaultPackageManager == nil {
+			return "", fmt.Errorf("package manager is not supported")
+		}
+		return software.DefaultPackageManager.PackageArchitecture()
+	},
+	"sys.pkg_type": func() (string, error) {
+		if software.DefaultPackageManager == nil {
+			return "", fmt.Errorf("package manager is not supported")
+		}
+		return string(software.DefaultPackageManager.Type()), nil
+	},
+	"sys.os": func() (string, error) {
+		systemInventory, err := inventory.CollectSystemInventory()
```

Comment:
I think we need to introduce some form of caching for the system inventory, as now it's possible that we will call it several times each agent interval (if multiple variables depending on it are used). On the other hand the system inventory collection is quite fast... What do you think @jonhenrik13 ?

Reply:
Yes, I was thinking about caching SystemInventory, but wanted to start the discussion first. I believe we should cache the things that are not expected to change during a scheduled run. This would need to happen at the early stages of agent execution, I think. I'll try to implement something.